airflow/models/taskinstance.py

```diff
@@ -2455,7 +2456,7 @@ def _run_raw_task(
 
             try:
                 if not mark_success:
-                    self._execute_task_with_callbacks(context, test_mode, session=session)
+                    result = self._execute_task_with_callbacks(context, test_mode, session=session)
```

Comment:
1. It seems that not all operators return a `result` on execution (example: [DatabricksSubmitRunOperator](https://github.com/apache/airflow/blob/main/airflow/providers/databricks/operators/databricks.py#L563-L564))
2. Some operators do return a `result` - like ([GlueOperator](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/glue.py#L217), [EmrServerlessStartJobOperator](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/emr.py#L1472)), but the return value is `job_id` in these scenarios.

Questions -
1. If any of the above operators publish an Airflow dataset, how to specify `extra` dictionary in the corresponding dataset?

Reply:
Regarding 1+2: Yes, if no result is generated, the content will be just `None`. If it is a scalar like a job_id it is rather a string. So in such cases the output is not usable for passing along to the dataset event.

Regarding your question 1: The publish of the event actually is happening _after_ this line of code. The change was attemping to capture the result to put is as extra a few lines of code later from here.